src/components/Button/index.js

```diff
@@ -161,94 +157,116 @@ const defaultProps = {
     forwardedRef: undefined,
 };
 
-class Button extends Component {
-    constructor(props) {
-        super(props);
-
-        this.renderContent = this.renderContent.bind(this);
-    }
-
-    componentDidMount() {
-        if (!this.props.pressOnEnter) {
-            return;
-        }
-
-        const shortcutConfig = CONST.KEYBOARD_SHORTCUTS.ENTER;
-
-        // Setup and attach keypress handler for pressing the button with Enter key
-        this.unsubscribe = KeyboardShortcut.subscribe(
-            shortcutConfig.shortcutKey,
-            (e) => {
-                if (!validateSubmitShortcut(this.props.isFocused, this.props.isDisabled, this.props.isLoading, e)) {
-                    return;
-                }
-                this.props.onPress();
-            },
-            shortcutConfig.descriptionKey,
-            shortcutConfig.modifiers,
-            true,
-            this.props.allowBubble,
-            this.props.enterKeyEventListenerPriority,
-            false,
-        );
-    }
-
-    componentWillUnmount() {
-        // Cleanup event listeners
-        if (!this.unsubscribe) {
-            return;
-        }
-        this.unsubscribe();
-    }
-
-    renderContent() {
-        if (this.props.children) {
-            return this.props.children;
+function Button({
+    allowBubble,
+    text,
+    shouldShowRightIcon,
+
+    icon,
+    iconRight,
+    iconFill,
+    iconStyles,
+    iconRightStyles,
+
+    small,
+    large,
+    medium,
+
+    isLoading,
+    isDisabled,
+
+    onPress,
+    onLongPress,
+    onPressIn,
+    onPressOut,
+    onMouseDown,
+
+    pressOnEnter,
+    enterKeyEventListenerPriority,
+
+    style,
+    innerStyles,
+    textStyles,
+
+    shouldUseDefaultHover,
+    success,
+    danger,
+    children,
+
+    shouldRemoveRightBorderRadius,
+    shouldRemoveLeftBorderRadius,
+    shouldEnableHapticFeedback,
+
+    id,
+    accessibilityLabel,
+    forwardedRef,
+}) {
+    const isFocused = useIsFocused();
+
+    const keyboardShortcutCallback = useCallback(
+        (event) => {
+            if (!validateSubmitShortcut(isFocused, isDisabled, isLoading, event)) {
+                return;
+            }
+            onPress();
+        },
+        [isDisabled, isFocused, isLoading, onPress],
+    );
+
+    useKeyboardShortcut(CONST.KEYBOARD_SHORTCUTS.ENTER, keyboardShortcutCallback, {
+        isActive: pressOnEnter,
+        shouldBubble: allowBubble,
+        priority: enterKeyEventListenerPriority,
+        shouldPreventDefault: false,
+    });
```

Comment:
This caused a regression https://github.com/Expensify/App/issues/33764. Previously we used to subscribe only once (`componentDidMount`). Now the `useKeyboardShortcut` hook re-runs based on dependencies. One of the dependencies is the `keyboardShortcutCallback`. This one also relies on other dependencies including `isFocused`. The isFocused value changes when the screen is unfocused/focused which causes the keyboard subscription to run again. When the keyboard subscription runs again it gets the highest priority and messes with other pages.

Explanation 
Before:
1. Mount page 1 -> Add keyboard listener for page 1  -> Mount page 2 -> Add keyboard listener for page 2
2. Pressing ENTER will trigger on page 2


After:
1. Mount page 1 -> Add keyboard listener for page 1  -> Mount page 2 -> Add keyboard listener for page 2 -> Page 1 is now unmouted -> Add keyboard listener for page 1 (since the useEffect triggers on page unfocus)
2. Pressing ENTER will trigger on page 1